### PR TITLE
Add h3-pyspark recipe

### PR DESCRIPTION
**Pull request by kevinschaich**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by kevinschaich** [comment]
@conda-forge/help-python Ready for review.

**Comment by kevinschaich** [comment]
@conda-forge-admin Ready for review

File: recipes/h3-pyspark/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/kevinschaich/h3-pyspark/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 64c39a66664676ce799dbfb5cbd49d9a9d76926e5495dcc8ea580fb03b4b46fb
+
+build:
+  noarch: python
+  number: 0
+  # skip: True  # [py<35]
```
Comment (beenje) [inline recipes/h3-pyspark/meta.yaml:14]: Please remove this comment

File: recipes/h3-pyspark/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (beenje) [inline recipes/h3-pyspark/meta.yaml:15]: Replace this with:

```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

File: recipes/h3-pyspark/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    - pyspark >=3.0.0
+    - h3-py >=3.7.0
+    - shapely >=1.7.0
+
+test:
+  requires:
+    - python >=3.6
+    - pyspark >=3.0.0
+    - h3-py >=3.7.0
+    - shapely >=1.7.0
```
Comment (beenje) [inline recipes/h3-pyspark/meta.yaml:32]: Please remove those requirements.

The goal of the test is to check that the package can be installed properly with all its needed dependencies (that you define in run requirements).
`requires` in test is only to install extra things that could be required to run tests.

**Review by beenje — CHANGES_REQUESTED** [review]

**Comment by beenje** [comment]
@conda-forge/help-python, this recipe looks good to me

**Review by dopplershift — APPROVED** [review]

**Pull request merged**